### DISCUSSION TITLE
Cannot query my model by component property

### DISCUSSION THREAD

**Discussion by Torone** [strapi/strapi#7910] [Legacy Topics]
You introduced Components and so I'm using it.

Now my simple model has a component called location and it store: city, lat and lng.

I created a custom controller to fetch data using the location attribute:

```
async findByCity(ctx) {
  const { city } = ctx.request.body
  const entities = await strapi.query('users').find({ 'location.city': 'London' })
  return entities.map(entity => sanitizeEntity(entity, { model: strapi.models.users }));
}
```

The message I receive is this: Your filters contain a field 'location.city' that doesn't appear on your model definition nor it's relations

But my model has the location, but it is a component:

```
"location": {
  "type": "component",
  "repeatable": false,
  "component": "profile.user-location"
},
```

So my question is: How can I query data based on components properties?

**Comment by derrickmehaffy** [comment]
Hello, currently it's not possible to filter on components: https://github.com/strapi/strapi/issues/5124

**Reply by Torone** [reply]
And so, what is the point to have the components? Looking at your answer they are totally useless by my point of view. Thanks for clarify it so I can stop wasting my time trying to do it and I will move it back to the model...

**Reply by derrickmehaffy** [reply]
I wouldn't say completely useless but I do agree their use-case is fairly limited. Generally what I see it for is structuring of a page or meta-data that you don't need to search through.

Models and relations are certainly the better choice in your case. :wink:

**Reply by Torone** [reply]
I see. I look forward to see the components enhanced, I think that it would be really useful to better structure your model in a less flat way. That's why I started using it. Unfortunately I also need to search and filter my data using those values. Thanks for your reply, very useful! Cheers
